backend/remote-state/s3/client.go

```diff
@@ -92,21 +95,33 @@ func (c *RemoteClient) Get() (payload *remote.Payload, err error) {
 }
 
 func (c *RemoteClient) get() (*remote.Payload, error) {
-	output, err := c.s3Client.GetObject(&s3.GetObjectInput{
-		Bucket: &c.bucketName,
-		Key:    &c.path,
-	})
+	var output *s3.GetObjectOutput
+	var err error
 
-	if err != nil {
-		if awserr := err.(awserr.Error); awserr != nil {
-			if awserr.Code() == "NoSuchKey" {
-				return nil, nil
-			} else {
-				return nil, err
+	// we immediately retry on an internal error, as those are usually transient
+	maxRetries := 2
+	for retryCount := 0; ; retryCount++ {
+		output, err = c.s3Client.GetObject(&s3.GetObjectInput{
+			Bucket: &c.bucketName,
+			Key:    &c.path,
+		})
+
+		if err != nil {
+			if awserr, ok := err.(awserr.Error); ok {
+				switch awserr.Code() {
+				case s3.ErrCodeNoSuchKey:
+					return nil, nil
+				case s3ErrCodeInternalError:
+					log.Println("[WARN] s3 internal error, retrying...")
```

Comment:
I think this log line belongs _after_ the `if` statement that follows, or else it'll print "retrying..." just before it exits.

Are there any interesting details in the error object that we should log here, or is this already being done by the AWS client itself?

Reply:
Of course, thanks for catching that.

The error itself is returned when we fail, but there's not much to get from any of them as they just contain some generic text. They are also logged through the aws client, so there's not much else to do here.